hat/app/org/hatdex/hat/api/controllers/RichData.scala

```diff
@@ -664,24 +654,30 @@ class RichData @Inject() (
     keyId: String): Future[Either[ContractVerificationFailure, ContractVerificationSuccess]] = {
     import ContractVerificationFailure._
     import ContractVerificationSuccess._
-    adjudicatorClient
-      .getPublicKey(contractRequestBodyRefined.contractId, contractRequestBodyRefined.hatName.value, keyId)
-      .map { response =>
-        response.status match {
-          case OK =>
+    import PublicKeyRequestSuccess._
+
+    adjudicatorClient.getPublicKey(contractRequestBodyRefined.hatName.value, contractRequestBodyRefined.contractId, keyId).map { publicKeyReqsponse =>
+      {
+        publicKeyReqsponse match {
+          case Left(PublicKeyRequestFailure.ServiceRespondedWithFailure(failureDescription)) => {
+            Left(ContractVerificationFailure.ServiceRespondedWithFailure(s"The Adjudicator Service responded with an error: ${failureDescription}"))
+          }
+          case Left(PublicKeyRequestFailure.InvalidPublicKeyFailure(failureDescription)) => {
+            Left(ContractVerificationFailure.ServiceRespondedWithFailure(s"The Adjudicator Service responded with an error: ${failureDescription}"))
+          }
+          case Right(PublicKeyReceived(publicKey)) => {
             val tryJwtClaim = ShortLivedTokenOps.verifyToken(
```

Comment:
i find nested case statements hard to read but potentially its a personal thing
in this case you could extract the logic into a method and pass the publicKey
give the method a good name and i know what it does i dont have to dig through the details